src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/SummaryJavadocCheck.java

```diff
@@ -384,4 +529,336 @@ private static String trimExcessWhitespaces(String text) {
+     * @return true, if first sentence contains allowed tags.
+     */
+    private static boolean validInlineTagInsideHtmlFormat(DetailNode ast) {
+        boolean contains = true;
+        if (getInlineTagNodeInsideHtml(ast) != null) {
+            for (DetailNode node : getInlineTagNodeInsideHtml(ast).getChildren()) {
+                if (node.getType() == JavadocTokenTypes.DESCRIPTION) {
+                    contains = checkInlineTagInsideDescription(node);
+                }
+            }
+        }
+        return contains;
+    }
+
+    /**
+     * Finds and return if the Inline Tag in description Node has allowed Inline Tags.
+     *
+     * @param description Description Node.
+     * @return true, if first sentence contains allowed tags.
+     */
+    private static boolean checkInlineTagInsideDescription(DetailNode description) {
+        boolean checks = true;
+        for (DetailNode nestedInlineTag : description.getChildren()) {
+            if (nestedInlineTag.getType() == JavadocTokenTypes.JAVADOC_INLINE_TAG) {
+                final DetailNode[] child = nestedInlineTag.getChildren();
+
+                // Checking size of child is not required, since child contains
+                // children of Inline Tag, as at least 2 children will be present which are
+                // RCURLY and LCURLY.
+                if (child[1].getType() == JavadocTokenTypes.CUSTOM_NAME
+                        && !ALLOWED_INLINE_TAGS.contains(child[1].getText())) {
+                    checks = false;
+                    break;
+                }
+            }
+        }
+        return checks;
+    }
+
+    /**
+     * Get Inline tag inside Inline tag from a javadoc node.
+     *
+     * @param javadoc Inline tag DetailNode.
+     * @return DetailNode with CUSTOM_NAME;
+     */
+    private static DetailNode getInlineTag(DetailNode... javadoc) {
+        DetailNode tagName = null;
+        if (getInlineTagNode(javadoc) != null) {
+            for (DetailNode child : getInlineTagNode(javadoc).getChildren()) {
+                if (child.getType() == JavadocTokenTypes.CUSTOM_NAME) {
+                    tagName = child;
+                }
+            }
+        }
+        return tagName;
+    }
+
+    /**
+     * Gets inline tag Node from a javadoc node.
+     *
+     * @param javadoc Inline tag DetailNode.
+     * @return DetailNode with CUSTOM_NAME;
+     */
+    private static DetailNode getInlineTagNode(DetailNode... javadoc) {
+        DetailNode tagNode = null;
+        for (DetailNode node : javadoc) {
+            if (node.getType() == JavadocTokenTypes.DESCRIPTION) {
+                for (DetailNode nodeChild : node.getChildren()) {
+                    if (nodeChild.getType() == JavadocTokenTypes.JAVADOC_INLINE_TAG) {
+                        tagNode = nodeChild;
+                    }
+                }
+            }
+        }
+        return tagNode;
+    }
+
+    /**
+     * Checks if period is at the end of sentence.
```

Comment:
it does not check anything, it return value

Reply:
Done